src/monodroid/jni/monodroid-glue.c

```diff
@@ -3510,27 +3632,27 @@ setup_environment_from_file (const char *apk, int index, int apk_count)
 }
 
 static void
-for_each_apk (JNIEnv *env, jobjectArray runtimeApks, void (*handler) (const char *apk, int index, int apk_count))
+for_each_apk (JNIEnv *env, jobjectArray runtimeApks, void *user_data, void (*handler) (const char *apk, int index, int apk_count, void *user_data))
```

Comment:
Isn't the `user_data` parameter conventionally the *last* parameter, e.g. [`g_qsort_with_data()`](https://developer.gnome.org/glib/stable/glib-Miscellaneous-Utility-Functions.html#g-qsort-with-data), [`g_list_foreach()`](https://developer.gnome.org/glib/stable/glib-Doubly-Linked-Lists.html#g-list-foreach)?

Rephrased, I'd expect:

```c
for_each_apk (JNIEnv *env, jobjectArray runtimeApks, void (*handler) (const char *apk, int index, int apk_count, void *user_data), void *user_data)
```